### PR TITLE
ethernet driver net_pkt api switch and bug fix

### PR DESCRIPTION
**Pull request by tbursztyka**

### CODE REVIEW

File: drivers/ethernet/eth_stellaris.c [hunk 1]
```diff
@@ -175,38 +183,34 @@ static int eth_stellaris_rx_pkt(struct device *dev, struct net_pkt *pkt)
...
 	struct net_if *iface = dev_data->iface;
-	int frame_len, ret;
+	struct net_pkt *pkt;
 
-	/* Obtain the packet to be populated. */
-	pkt = net_pkt_get_reserve_rx(K_NO_WAIT);
+	pkt = eth_stellaris_rx_pkt(dev, iface);
 	if (!pkt) {
-		LOG_ERR("Could not allocate pkt");
+		LOG_ERR("Failed to read data");
```
Comment (jukkar) [inline drivers/ethernet/eth_stellaris.c]: Minor thing, but I think we should say "Failed to write data" here.

File: drivers/ethernet/eth_smsc911x.c [hunk 2]
```diff
@@ -578,12 +578,13 @@ static struct net_pkt *smsc_recv_pkt(struct dehive *dev, u32_t pkt_size)
 
 	/* Discard FCS */
 	{
-		u32_t dummy = SMSC9220->RX_DATA_PORT;
+		u32_t __unused dummy = SMSC9220->RX_DATA_PORT;
```
Comment (jukkar) [inline drivers/ethernet/eth_smsc911x.c]: Can't we just remove the dummy block here?

Comment (tbursztyka) [reply drivers/ethernet/eth_smsc911x.c]: No, this line does flush the rx data port register by reading it. Original code needed it, so I guess this is still valid.

File: drivers/ethernet/eth_stellaris.c (hunk 1 above)
Comment (tbursztyka) [reply drivers/ethernet/eth_stellaris.c]: at this point we are actually trying to read data from the hw, sure we write it back to pkt. But it's same same I would say.

File: drivers/ethernet/eth_smsc911x.c (hunk 2 above)
Comment (jukkar) [reply drivers/ethernet/eth_smsc911x.c]: Ah, ok then.

**Review by rveerama1 — APPROVED** [review]

**Review by jukkar — APPROVED** [review]
LGTM

**Pull request merged**